webproject/src/main/webapp/js/chat-box.js

```diff
@@ -24,24 +27,42 @@ $(document).ready(function() {
 });
 
 function sendMessage() {
-    let messageBox = document.getElementById("message");
-    var message = messageBox.value;
-    messageBox.value = "";
-    var date = new Date();
-    var time = hours_with_leading_zeroes(date) + ":" + minutes_with_leading_zeroes(date);
-    firebase.database().ref('messages/' + roomID).push().set({
-        type: "text",
-        user: name,
-        message: message,
-        time: time
-    }, function(error) {
-        if (error) {
-            console.log("Write failed");
-        }
+    let usernamePromise = null;
+    
+    if (username == null) {
+	    usernamePromise = getUsername();
+    } else {
+	    usernamePromise = Promise.resolve(username);
+    }
+
+    usernamePromise.then(username => {
+        let messageBox = document.getElementById("message");
+        var message = messageBox.value;
+        messageBox.value = "";
+        var date = new Date();
+        var time = hours_with_leading_zeroes(date) + ":" + minutes_with_leading_zeroes(date);
```

Comment:
this should be unix timestamp, instead of just time. since you need to consider users who is not in singapore.
but lets fix this later

Reply:
Understood!